File: src/main/java/catwalks/util/meta/BoolProp.java
package catwalks.util.meta;

/**
 * Created by TheCodeWarrior
 */
public class BoolProp extends MetaValueProperty<Boolean> {
	
	protected int index;
	
	public BoolProp(int index) {
		this.index = index;
	}
	
	@Override
	public boolean addBits(int index, int bits) {
		return false;
	}
	
	@Override
	public void set(MetaStorage storage, Boolean value) {
		storage.set(index, value);
		storage.notifyIfDirty();
	}
	
	@Override
	public Boolean get(MetaStorage storage) {
		return storage.get(index);
	}
}

File: src/main/java/catwalks/util/meta/BoolProp.kt
package catwalks.util.meta

/**
 * Created by TheCodeWarrior
 */
class BoolProp(protected var index: Int) : MetaValueProperty<Boolean>() {

    override fun addBits(index: Int, bits: Int): Boolean {
        return false
    }

    override fun set(storage: MetaStorage, value: Boolean) {
        storage.set(index, value)
        storage.notifyIfDirty()
    }

    override fun get(storage: MetaStorage): Boolean {
        return storage.get(index)
    }
}
